core/src/main/java/org/bitcoinj/wallet/KeyChain.java

```diff
@@ -76,11 +77,19 @@ enum KeyPurpose {
     int numBloomFilterEntries();
 
     /**
-     * <p>Returns the earliest creation time of keys in this chain, in seconds since the epoch. This can return zero
-     * if at least one key does not have that data (was created before key timestamping was implemented). If there
-     * are no keys in the wallet, {@link Long#MAX_VALUE} is returned.</p>
+     * Returns the earliest creation time of keys in this chain.
+     * @return earliest creation times of keys in this chain,
+     *         {@link Instant#EPOCH} if at least one time is unknown,
+     *         {@link Instant#MAX} if no keys in this chain
      */
-    long getEarliestKeyCreationTime();
+    Instant getEarliestKeyCreationTimeInstant();
+
+    /** @deprecated use {@link #getEarliestKeyCreationTimeInstant()} */
+    @Deprecated
+    default long getEarliestKeyCreationTime() {
+        Instant earliestKeyCreationTime = getEarliestKeyCreationTimeInstant();
+        return earliestKeyCreationTime.equals(Instant.MAX) ? Long.MAX_VALUE : earliestKeyCreationTime.getEpochSecond();
```

Comment:
Helper method for conversion. Multi-line ternary.

Reply:
I get your point. Since this case, and the other two occurences, will go away with removing the deprecated methods, I actually prefer this little bit of duplicated code. A helper method will be forgotten to remove, unless marked as deprecated to begin with (which will create deprecation warnings).